### PR TITLE
Add Job Schedule validation changes

### PR DESCRIPTION
**Pull request by ct-ajovanovic**
Kapua GUI Freezes If Schedule's Time Parameter Not In Proper Format #1169
Signed-off-by: ct-ajovanovic <aleksandra.jovanovic@comtrade.com>

### CODE REVIEW

**Comment by LeoNerdoG** [comment]
I have verified that this PR fixes issue #1169 .

**Comment by muros-ct** [comment]
@lorthirk Can you check this, it is a biger change. It introduces time validator.
I am concerned about fixed format for date.

File: console/module/job/src/main/java/org/eclipse/kapua/app/console/module/job/client/schedule/JobScheduleAddDialog.java [hunk 1]
```diff
@@ -87,9 +94,11 @@ public void createBody() {
 
         endsOn.setFieldLabel(JOB_MSGS.dialogAddScheduleEndsOnLabel());
         endsOn.setFormatValue(true);
+        endsOn.setAllowBlank(false);
```
Comment (lorthirk) [inline console/module/job/src/main/java/org/eclipse/kapua/app/console/module/job/client/schedule/JobScheduleAddDialog.java]: Why is this required? A schedule is not required to have an ending, meaning that it will be valid forever. Is this case covered somewhere else?

File: console/module/job/src/main/java/org/eclipse/kapua/app/console/module/job/client/schedule/JobScheduleAddDialog.java [hunk 2]
```diff
@@ -87,9 +94,11 @@ public void createBody() {
...
         endsOn.getPropertyEditor().setFormat(DateTimeFormat.getFormat("dd/MM/yyyy"));
         mainPanel.add(endsOn);
 
+        endsOnTime.setAllowBlank(false);
```
Comment (lorthirk) [inline console/module/job/src/main/java/org/eclipse/kapua/app/console/module/job/client/schedule/JobScheduleAddDialog.java]: Why is this required? A schedule is not required to have an ending, meaning that it will be valid forever. Is this case covered somewhere else?

**Comment by ct-ajovanovic** [comment]
Hello @lorthirk 
I created validation for startsOn and startsOnTime, following the same principle as in endsOn and endsOnTime. The mentioned two lines were added as a way of handling NullPointerException thrown when when the user tries to submit the new schedule without first defining endsOn and endsOnTime, after these changes. I see my mistake now, not taking in consideration that end date and time don't have to be defined. I will modify this.

**Comment by lorthirk** [comment]
So @muros-ct,

Apart from the two `setRequired(true)` I already reported to @ct-ajovanovic the code looks ok. However I'd like @LeoNerdoG to have some UI tests on the feature, just to make sure it still performs as expected, before merging.

**Comment by ct-ajovanovic** [comment]
Hi @lorthirk 
I made the necessary changes regarding endsOn and endsOnTime fields, and date and time validation. The validation messages were also changed, as recommended by @LeoNerdoG .
Can you please review this?

**Comment by LeoNerdoG** [comment]
I have verified that this PR successfully resolves issue #1169 .

**Comment by muros-ct** [comment]
@LeoNerdoG there were some minor changes, please re-evaluate.
If ok we can merge.

**Comment by LeoNerdoG** [comment]
@muros-ct Everything OK, it can be merged.
